src/bp/ui-studio/src/web/views/OneFlow/sidePanel/TopicList/index.tsx

```diff
@@ -132,305 +136,244 @@ const TopicList: FC<Props> = props => {
             icon="edit"
-            text={lang.tr('edit')}
+            label={lang.tr('edit')}
             onClick={() => props.editQnA(name.replace('/qna', ''))}
           />
-        </Menu>
+        </Fragment>
       )
-    } else if (elementType === 'document') {
+    } else {
       const { name } = element as NodeData
 
       return (
-        <Menu>
+        <Fragment>
           <MenuItem
             id="btn-edit"
             disabled={props.readOnly}
-            icon="edit"
-            text={lang.tr('studio.flow.topicList.editWorkflow')}
-            onClick={() => props.editWorkflow(name, element)}
+            label={lang.tr('studio.flow.sidePanel.renameWorkflow')}
+            onClick={() => setEditing(path)}
           />
           <MenuItem
             id="btn-duplicate"
             disabled={props.readOnly}
-            icon="duplicate"
-            text={lang.tr('duplicate')}
+            label={lang.tr('duplicate')}
             onClick={() => props.duplicateFlow(name)}
           />
-          <MenuItem
-            id="btn-export"
-            disabled={props.readOnly}
-            icon="export"
-            text={lang.tr('export')}
-            onClick={() => props.exportWorkflow(name)}
-          />
-          <MenuDivider />
           <MenuItem
             id="btn-delete"
             disabled={lockedFlows.includes(name) || !props.canDelete || props.readOnly}
-            icon="delete"
-            text={lang.tr('delete')}
+            label={lang.tr('delete')}
+            intent={Intent.DANGER}
             onClick={() => deleteFlow(name)}
           />
-        </Menu>
+        </Fragment>
       )
     }
   }
 
-  const nodeRenderer = (el: NodeData) => {
-    const { name, label, icon, type, triggerCount, referencedIn, countByTopic } = el
-    const displayName = label || name.substr(name.lastIndexOf('/') + 1).replace(/\.flow\.json$/, '')
-    const isFocused = name === props.focusedText
-    const isNew = name === props.newPath
-    const isQna = type === 'qna'
-    const filterOrder = isNew ? 3 : !isQna ? 2 : 1
-
-    const editWorkflow = async newName => {
-      props.setFocusedText(undefined)
-      props.setNewPath(undefined)
-      setForceSelect(undefined)
-      if (newName === '') {
-        await props.fetchFlows()
-        await props.fetchTopics()
-      } else if (newName !== displayName) {
-        const fullName = buildFlowName({ topic: el.topic, workflow: sanitize(newName) }, true)
-        if (!props.flowsName.find(x => x.name === fullName)) {
-          props.renameFlow({ targetFlow: name, name: fullName })
-          props.updateFlow({ name: fullName })
-          setForceSelect({ field: 'fullPath', value: fullName })
-        } else {
-          await props.fetchFlows()
-          await props.fetchTopics()
+  const newFlows = {}
+
+  for (const workflow of flows) {
+    const splitPath = workflow.name.split('/')
+    const nodeLabel = splitPath.pop().replace('.flow.json', '')
```

Comment:
Can you add a todo to refactor this part when that PR will be merged ? There is an helper method to extract each parts of a workflow, since we often need to extract part of it.

 https://github.com/botpress/botpress/pull/3272/files#diff-13a273103517c41a0c7dfec1c06f75b3